esrally/mechanic/telemetry.py

```diff
@@ -191,7 +191,7 @@ def instrument_java_opts(self, car, candidate_id):
 
     def java_opts(self, log_file):
         recording_template = self.telemetry_params.get("recording-template")
-        java_opts = ["-XX:+UnlockDiagnosticVMOptions", "-XX:+DebugNonSafepoints"]
+        java_opts = ["-XX:+DebugNonSafepoints", "-XX:+UnlockDiagnosticVMOptions"]
```

Comment:
If the order is preserved (see my comment above) also this needs to be reversed again (`-XX:+UnlockDiagnosticVMOptions` must be specified before `-XX:+DebugNonSafepoints`).